### ISSUE TITLE
[a11y] Add aria-live for screen reader announcements

### ISSUE THREAD

**Issue by jsmrtn** [putyourlightson/craft-sprig#48]
ARIA can flag dynamic content updates so that screen readers announce when content has been updated (see [W3 WAI-ARIA guidelines](https://www.w3.org/TR/wai-aria-1.1/#aria-live)).

I propose `aria-live="polite"` is added to the wrapper div injected by sprig to allow better use on screen readers. It is my understanding it should be added to the injected div rather than the content within, because the div is the only immutable element (as the inner divs will be replaced entirely).

`polite` is more suitable than `assertive` in this instance as WAI-ARIA recommend you use `polite` for things like updating content:

>aria-live="polite"
Any updates made to this region should only be announced if the user is not currently doing anything. live="polite" should be used in most situations involving live regions that present new information to users, such as updating news headlines.

It might also be worth adding `aria-atomic="true"`, but I appreciate there are instances where you are using sprig to read more, rather than for pagination or wholesale content changes.

**Comment by bencroker** [comment]
Thanks for the suggestion! I wondered previously whether Sprig should implement something along the lines of what you are proposing, however I decided to leave it to the developer to dictate how ARIA attributes should be used. This is easily achieved with a wrapping element:

```twig
{# In the component #}
<div aria-live="polite">
    {# Output updated content #}
</div>
```

**Comment by jsmrtn** [comment]
So if I have the following structure...

*Template*
```
{{ sprig('component/_listing.twig', {
    offset: 0,
    limit: limit
}) }}
```

*component/_listing.twig* 
```
<div class="listing">
    [..]
</div>
```

When this is compiled, it gets outputted like this...
```
<div id="component-yeurhz" class="sprig-component" hx-target="this" hx-include="#component-yeurhz *" hx-trigger="refresh" hx-get="" hx-vars="">
    <div class="listing">
        [..]
    </div>
</div>
```

So if I am putting the `aria-live="polite"` on the `<div class="listing">`, wouldn't that entire div be replaced? Hence the `aria-live="polite"` should be on the auto-generated div–me adding a wrapper div in the component would be redundant as it would be replaced itself.

Is there a setting I should add to get the `<div class="listing">` to have its content replaced, rather than being replaced itself?

**Comment by bencroker** [comment]
> So if I am putting the aria-live="polite" on the `div class="listing"`, wouldn't that entire div be replaced? 

Yes it would, but is that an issue assuming that only the contents of that div could change?

Alternatively, it is already possible to put attributes on the component using the third parameter, as follows:

```twig
{{ sprig('component/_listing.twig', { }, {'aria-live': 'polite'}) }}
```

See https://putyourlightson.com/plugins/sprig#component-attributes

**Comment by jsmrtn** [comment]
> Alternatively, it is already possible to put attributes on the component using the third parameter, as follows:
> 
> ```
> {{ sprig('component/_listing.twig', { }, {'aria-live': 'polite'}) }}
> ```
> 
> See https://putyourlightson.com/plugins/sprig#component-attributes

This looks like it is exactly what I was after! Thanks Ben.

**Comment by robzor** [comment]
Sorry to thread ressurrrect, but if I wanted to change the `aria-busy` attribute here, what would be the way to do that?

i.e. can we change it to `aria-busy="true"` while sprig is running the process? Is that possible? I know there is s-indicator but this shows/hides an element, afaik?

```
{# Sprig: News list #}
    {{ sprig("__sprig/news-list", {
      _limit: 15,
      _slug: newsPageSlug
    }, {
      'aria-live': 'polite',
      'aria-atomic': 'true',
      'aria-busy': 'false'
    }) }}
```

Any help you can give would be appreciated!

Thanks,

Robin

**Comment by bencroker** [comment]
The Sprig component element is not replaced during a component refresh, so I don’t think there’s any trivial way to accomplish this. Please create a new issue if you’d like to discuss further.
